kernel/mutex.c

```diff
@@ -82,12 +82,9 @@ void _impl_k_mutex_init(struct k_mutex *mutex)
 }
 
 #ifdef CONFIG_USERSPACE
-u32_t _handler_k_mutex_init(u32_t mutex, u32_t arg2, u32_t arg3,
-			    u32_t arg4, u32_t arg5, u32_t arg6, void *ssf)
+_SYSCALL_HANDLER1(k_mutex_init, mutex)
```

Comment:
k_mutex syscalls could be _SIMPLE_VOID.

Reply:
k_mutex_lock takes an additional timeout parameter